.pre-commit-config.yaml

```diff
@@ -0,0 +1,10 @@
+repos:
+-   repo: https://github.com/pycqa/flake8
+    rev: 7.1.1
+    hooks:
+    -   id: flake8
```

Comment:
I need to try running it again, but I got an error on trying to commit from a Windows system. I clicked through the error, but it was something about flake8  not being configured? I'll rebuild  my venv and see what I get.

Reply:
I tested by creating a branch, making a line of circuit.py that was long enough for black to reformat, and trying to commit.

Like making this all one line instead of having line feeds in it: https://github.com/C2QA/bosonic-qiskit/blob/main/c2qa/circuit.py#L1046-L1054

Reply:
Maybe it is normal for pre-commit to not format & commit black? I'm familiar with npm and having it use pre-commit hooks format & commit automatically for you instead of having to try committing twice.